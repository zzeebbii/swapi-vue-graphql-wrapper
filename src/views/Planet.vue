<template>
  <div>
    <div class="error" v-if="error">{{ error }}</div>
    <div v-if="loading" class="loader"></div>
    <div class="info" v-if="planet">
      <table>
        <tbody>
          <tr>
            <td>Name</td>
            <td>{{ planet.name }}</td>
          </tr>
          <tr>
            <td>Climate</td>
            <td>{{ planet.climate }}</td>
          </tr>
          <tr>
            <td>Diameter</td>
            <td>{{ planet.diameter }}</td>
          </tr>
          <tr>
            <td>Gravity</td>
            <td>{{ planet.gravity }}</td>
          </tr>
          <tr>
            <td>Orbital Period</td>
            <td>{{ planet.orbital_period }}</td>
          </tr>
          <tr>
            <td>Population</td>
            <td>{{ planet.population }}</td>
          </tr>
          <tr>
            <td>People</td>
            <td v-if="planet.people">
              <ul>
                <li v-for="person in planet.people" :key="person.name">{{ person.name }}</li>
              </ul>
            </td>
          </tr>
          <tr>
            <td>Rotation Period</td>
            <td>{{ planet.rotation_period }}</td>
          </tr>
          <tr>
            <td>Surface Water</td>
            <td>{{ planet.surface_water }}</td>
          </tr>
          <tr>
            <td>Terrain</td>
            <td>{{ planet.terrain }}</td>
          </tr>
          <tr>
            <td>URL</td>
            <td>{{ planet.url }}</td>
          </tr>
          <tr>
            <td>Created</td>
            <td>{{ planet.created }}</td>
          </tr>
          <tr>
            <td>Edited</td>
            <td>{{ planet.edited }}</td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</template>

<script>
// @ is an alias to /src

import { SINGLE_PLANET_QUERY } from "@/utils/constants";

export default {
  name: "Planet",
  data() {
    return {
      planet: null,
      loading: 0,
      id: this.$route.params.id,
      error: null
    };
  },
  apollo: {
    planet: {
      query: SINGLE_PLANET_QUERY,
      variables() {
        return {
          id: this.id
        }
      },
      error(error) {
        this.error = JSON.stringify(error.message);
      }
    }
  }
};
</script>
