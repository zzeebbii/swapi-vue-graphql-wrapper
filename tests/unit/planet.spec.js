import { shallowMount, createLocalVue } from "@vue/test-utils";
import VueRouter from "vue-router";
import Planet from "@/views/Planet.vue";

const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();

describe("Planet.vue", () => {
  it("should render with data", function() {
    const PLANET_DATA = {
      climate: "Cold",
      created: "1990",
      diameter: 40000,
      edited: "2010",
      gravity: "Normal",
      name: "Mars",
      orbital_period: "2 Year",
      population: 1000000,
      people: [],
      rotation_period: 20000,
      surface_water: "Normal",
      terrain: "Plain",
      url: "http://swapi.co"
    };

    const wrapper = shallowMount(Planet, {
      localVue,
      router,
      data() {
        return {
          planet: PLANET_DATA
        };
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
