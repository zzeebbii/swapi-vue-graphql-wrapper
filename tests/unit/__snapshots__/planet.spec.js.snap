// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet.vue should render with data 1`] = `
<div>
  <!---->
  <!---->
  <div class="info">
    <table>
      <tbody>
        <tr>
          <td>Name</td>
          <td>Mars</td>
        </tr>
        <tr>
          <td>Climate</td>
          <td>Cold</td>
        </tr>
        <tr>
          <td>Diameter</td>
          <td>40000</td>
        </tr>
        <tr>
          <td>Gravity</td>
          <td>Normal</td>
        </tr>
        <tr>
          <td>Orbital Period</td>
          <td>2 Year</td>
        </tr>
        <tr>
          <td>Population</td>
          <td>1000000</td>
        </tr>
        <tr>
          <td>People</td>
          <td>
            <ul></ul>
          </td>
        </tr>
        <tr>
          <td>Rotation Period</td>
          <td>20000</td>
        </tr>
        <tr>
          <td>Surface Water</td>
          <td>Normal</td>
        </tr>
        <tr>
          <td>Terrain</td>
          <td>Plain</td>
        </tr>
        <tr>
          <td>URL</td>
          <td>http://swapi.co</td>
        </tr>
        <tr>
          <td>Created</td>
          <td>1990</td>
        </tr>
        <tr>
          <td>Edited</td>
          <td>2010</td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
